src/textual/_segment_tools.py

```diff
@@ -14,6 +14,47 @@
 from .geometry import Size
 
 
+def index_to_cell_position(segments: Iterable[Segment], index: int) -> int:
+    """Given a character index, return the cell position of that character within
+    an Iterable of Segments. This is the sum of the cell lengths of all the characters
+    *before* the character at `index`.
+
+    Args:
+        segments (Iterable[Segment]): The segments to find the cell position within.
+        index (int): The index to convert into a cell position.
+
+    Returns:
+        int: The cell position of the character at `index`. If the `index` is not
+            valid within the segments, returns 0.
+    """
+    if index == 0:
+        return 0
+
+    cell_position_end = 0
+    segment_length = 0
+    segment_end_index = 0
+    segment_cell_length = 0
+    text = ""
+    iter_segments = iter(segments)
+    try:
+        while segment_end_index < index:
+            segment = next(iter_segments)
+            text = segment.text
+            segment_length = len(text)
+            segment_cell_length = cell_len(text)
+            cell_position_end += segment_cell_length
+            segment_end_index += segment_length
+    except StopIteration:
+        return 0
```

Comment:
Is `0` the expected return if index is not in segments?

My gut feeling is this should either be the total cell position or an error.

Reply:
I think an error - I could raise `IndexError`?